### PR TITLE
Add beta tag build trigger.

### PR DESCRIPTION
**Pull request by mberhault**
This _should_ go through the same motion as a master merge, with the
exception that it will only push two binaries two AWS.
given the tag beta-20160329, it will push:
binaries.cockroachdb.com/cockroach-beta-20160329.linux-amd64.tgz
which expands into:
cockroach-beta-20160329.linux-amd64/cockroach

I've tested the push-tagged-aws.sh part. I still need to test the
circle.yml part and make sure the right tag gets inserted into the
binary's Build Tag.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5689)

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -50,3 +50,17 @@ deployment:
...
       - build/push-aws.sh
+
+deployment:
+  docker:
+    tag: /beta-[0-9]+/
+    commands:
+      - aws configure set region us-east-1
+      - build/build-static-binaries.sh
+      - mkdir -p "${CIRCLE_ARTIFACTS}/acceptance_deploy"
+      - time acceptance/acceptance.test -test.v -test.timeout 5m
```
Comment (mberhault) [inline circle.yml]: question: should we still run the acceptance test here?

**Comment by mberhault** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[circle.yml, line 56 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5689#-KE3C3Ebxd_2cMpheAUt:-KE3DAoHzRw2gMf9bifw:577371562) ([raw file](https://github.com/cockroachdb/cockroach/blob/acaaf4a0106fafc7b301ff498029b7d5caa7a758/circle.yml#L56)):*
yeah, I didn't expect it to exist, but I randomly searched for it during my smoke break. this seems like a much nicer way of doing it. releasing should be just a matter of running `git tag beta-20160329 && git push beta-20160329` and waiting ages

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5689)*

**Comment by petermattis** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[circle.yml, line 56 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5689#-KE3C3Ebxd_2cMpheAUt:-KE3EWmdqJKcbjO5QMVw:-866617342) ([raw file](https://github.com/cockroachdb/cockroach/blob/acaaf4a0106fafc7b301ff498029b7d5caa7a758/circle.yml#L56)):*
From the docs, it appears you can also tag a release on github which would have the same effect.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5689)*

**Comment by mberhault** [comment]
true. put a "release process" somewhere once I figure it out.
for now, I pulled master and ran:
```
git tag beta-20160328
git push origin beta-20160328
```

This has triggered a new build: https://circleci.com/gh/cockroachdb/cockroach/15503

**Comment by mberhault** [comment]
ugh.  acceptance tests aren't working for some reason:
```
time acceptance/acceptance.test -test.v -test.timeout 5m -i cockroachdb/cockroach -nodes 3 -l "${CIRCLE_ARTIFACTS}"/acceptance_deploy 2>&1 > "${CIRCLE_ARTIFACTS}/acceptance_deploy.log"
sys	0m0.008s
I160329 23:11:03.006152 rand.go:86  Random seed: -8532244621296312315
I160329 23:11:03.093272 acceptance/cluster/localcluster.go:279  Initializing Cluster AdHoc 3x1:
{"name":"AdHoc 3x1","nodes":[{"count":3,"stores":[{"count":1,"max_ranges":0}]}],"duration":5000000000}
I160329 23:11:03.096110 acceptance/cluster/localcluster.go:639  stopping
panic: Error: No such image: cockroachdb/cockroach [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x755a9d]
```

Any ideas?

**Comment by petermattis** [comment]
Lets skip running the acceptance tests for now. We'll figure out the
problem this week.

**Comment by mberhault** [comment]
we do still run the ones as part of the regular tests. does the one in the
deployment step use a just-built docker image or something?

**Comment by mberhault** [comment]
ok. after a very flaky circle-ci, we finally have binaries with just `git tag beta-xxx && git push origin beta-xxx`

This resulted in:
```
$ curl -O https://binaries.cockroachdb.com/cockroach-beta-20160101.darwin-10.9-amd64.tgz
$ tar xfz cockroach-beta-20160101.darwin-10.9-amd64.tgz
$ cockroach-beta-20160101.darwin-10.9-amd64/cockroach version
Build Vers:  go1.6
Build Tag:   beta-20160101
Build Time:  2016/03/30 01:39:37
Build Deps:
  github.com/biogo/store                     3b4c041f52c224ee4a44f5c8b150d003a40643a0
  github.com/chzyer/readline                 30b462e50b8f8a1172c3b2f780894c0db2ae2dcb
  github.com/cockroachdb/cmux                112f0506e7743d64a6eb8fedbcff13d9979bbf92
  github.com/cockroachdb/cockroach           7d356a93316165f36c3f772f80caabaa2dd29749
etc...
```

ditto for linux.

**Comment by cuongdo** [comment]
Why is the build tag `beta-20160101`? Are the numbers supposed to be the
date of the build or is that just coincidental?

**Comment by petermattis** [comment]
I'm guessing that was Marc testing. Marc, let's add a correct tag at the
same sha as beta-20160101 and call that our first beta release.

**Comment by mberhault** [comment]
indeed, that was just for testing, I wasn't sure how many I was going to need.

nothing else we absolutely must have?

**Comment by petermattis** [comment]
Nope. Let's cut it.

**Comment by mberhault** [comment]
ok. building beta-20160329
also went ahead and cleaned up all the other tags.

**Comment by vivekmenezes** [comment]
Congrats team!

**Comment by mberhault** [comment]
built and pushed:
https://binaries.cockroachdb.com/cockroach-beta-20160329.linux-amd64.tgz
https://binaries.cockroachdb.com/cockroach-beta-20160329.darwin-10.9-amd64.tgz

**Pull request merged**